x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/StartTrainedModelDeploymentAction.java

```diff
@@ -120,9 +124,29 @@ public String toString() {
 
     public static class TaskParams implements PersistentTaskParams, MlTaskParams {
 
-        public static final Version VERSION_INTRODUCED = Version.V_8_0_0;
+        // TODO add support for other roles? If so, it may have to be an instance method...
+        // NOTE, whatever determines allocation should not be dynamically set on the node
+        // Otherwise allocation logic might fail
+        public static boolean canAllocateToNode(DiscoveryNode node) {
```

Comment:
I wonder if we can find a name for this method that makes less of a claim. The reason is that this is often called before checking if the node has enough capacity to fit a model. However, as `canAllocateToNode` is too generic, it could mislead a caller to assume this encompasses the capacity check.

Reply:
`mayAllocateToNode` it is :D

Reply:
Perhaps something that explains the check is really about the node role? For instance, `isNodeInferenceCapable`?